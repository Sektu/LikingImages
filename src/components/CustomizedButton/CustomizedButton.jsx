import "./CustomizedButton.css";

const CustomizedButton = ({className, ariaLabel, name, onClick}) => {
    return (<button 
        className={className}
        aria-label={ariaLabel}
        onClick={onClick}>
            {name}
        </button>
        );
}

export default CustomizedButton;