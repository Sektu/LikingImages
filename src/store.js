import { configureStore } from "@reduxjs/toolkit";
import likingReducer from "./slices/likingSlice";
import imagesReducer from "./slices/imagesSlice";

export default configureStore({
  reducer: {
    liking: likingReducer,
    images: imagesReducer,
  }
});
