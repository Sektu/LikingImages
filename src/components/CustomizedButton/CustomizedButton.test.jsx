import React from 'react';
import { render, cleanup } from '@testing-library/react';
import store from "../../store";
import { Provider } from "react-redux";
import CustomizedButton from './CustomizedButton';

afterEach(cleanup);

it("renders", () => {
    const className = "klasunia";
    const ariaLabel = "do-do-like";
    const name = "nazwa";

    const { asFragment } = render(
        <Provider store={store}>
            <CustomizedButton className={className} ariaLabel={ariaLabel} name={name} />
        </Provider>);
    expect(asFragment()).toMatchSnapshot();
});

it("check customized button ", () => {
    const className = "klasunia";
    const ariaLabel = "do-do-like";
    const name = "nazwa";

    const { getByRole } = render(
        <Provider store={store}>
            <CustomizedButton className={className} ariaLabel={ariaLabel} name={name} />
        </Provider>);
    const button = getByRole("button");
    expect(button).toHaveClass(className);
    expect(button).toHaveAttribute('aria-label', ariaLabel);
    expect(button).toHaveTextContent(name);
});
