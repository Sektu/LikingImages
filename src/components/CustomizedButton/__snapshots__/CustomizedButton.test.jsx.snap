// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <button
    aria-label="do-do-like"
    class="klasunia"
  >
    nazwa
  </button>
</DocumentFragment>
`;
